src/uwtools/rocoto.py

```diff
@@ -143,3 +65,249 @@ def validate_rocoto_xml(input_xml: OptionalPath) -> bool:
+        # insert !DOCTYPE block, then write final XML.
+        xml = etree.tostring(
+            self._tidy(self._root), pretty_print=True, encoding="utf-8", xml_declaration=True
+        ).decode()
+        xml = re.sub(r"&amp;([^;]+);", r"&\1;", xml)
+        xml = self._insert_doctype(xml)
+        with writable(path) as f:
+            f.write(xml.strip())
+
+    @property
+    def _doctype(self) -> Optional[str]:
+        """
+        Generate the <!DOCTYPE> block with <!ENTITY> definitions.
+
+        :return: The <!DOCTYPE> block if entities are defined, otherwise None.
+        """
+        if entities := self._config[STR.workflow].get(STR.entities):
+            tags = (f'  <!ENTITY {k} "{v}">' for k, v in entities.items())
+            return "<!DOCTYPE workflow [\n%s\n]>" % "\n".join(tags)
+        return None
+
+    def _config_validate(self, config_file: OptionalPath) -> None:
+        """
+        Validate the given YAML config.
+
+        :param config_file: Path to the YAML config (defaults to stdin).
+        """
+        if not validate_yaml(
+            config_file=config_file, schema_file=resource_pathobj("rocoto.jsonschema")
+        ):
+            raise UWConfigError("YAML validation errors identified in %s" % config_file)
+
+    def _add_metatask(self, e: Element, config: dict, taskname: str) -> None:
+        """
+        Add a <metatask> element to the <workflow>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        :param taskname: The name of the metatask being defined.
+        """
+        e = SubElement(e, STR.metatask, name=taskname)
+        for key, val in config.items():
+            if key.startswith(STR.metatask):
+                self._add_metatask(e, val, self._tag_name(key)[1])
+            elif key.startswith(STR.task):
+                self._add_task(e, val, self._tag_name(key)[1])
+            elif key == STR.var:
+                for name, value in val.items():
+                    SubElement(e, STR.var, name=name).text = value
+
+    def _add_task(self, e: Element, config: dict, taskname: str) -> None:
+        """
+        Add a <task> element to the <workflow>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        :param taskname: The name of the task being defined.
+        """
+        e = SubElement(e, STR.task, name=taskname)
+        self._set_attrs(e, config)
+        # These elements are simple enough to not require their own methods:
+        for name in (STR.account, STR.command, STR.nodes, STR.walltime):
+            if name in config:
+                SubElement(e, name).text = config[name]
+        # The job name, if unspecified, defaults to the task name.
+        SubElement(e, STR.jobname).text = config.get(STR.jobname, taskname)
+        # Add any dependencies:
```

Comment:
It doesn't matter what we do here because the output will be sorted by element anyway. I _really do not_ want to get into trying to customize output order.

Reply:
But if, depending on conversation in a different threads, it turns out the we cannot tidy the output for functional reasons, though that would be unfortunate, I will move the dependencies last.